chalk-solve/src/clauses/builtin_traits/copy.rs

```diff
@@ -54,6 +54,24 @@ pub fn add_copy_program_clauses<I: Interner>(
             TypeName::FnDef(_) => {
                 builder.push_fact(trait_ref.clone());
             }
+            TypeName::Closure(_) => {
+                let interner = db.interner();
+                match substitution
```

Comment:
I think we should just require that `substitution.parameters(interner).last`, or even `substitution.parameters(interner)`, be `copy` (or clone, etc) without bothering to unpack that it is a tuple. This is the direction rustc is going as well, for various reasons.

Reply:
Actually that is kind of nice.